warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,329 @@
+                } else {
+                    startDate = DateHelper.parse(it.next().getYyyymmdd());
+                    if (it.hasNext())
+                        endDate = DateHelper.parse(it.next().getYyyymmdd());
+                    else {
+                        endDate = originalEndDate;
+                    }
+                }
+                
+                results = getQueryData((ShardQueryConfiguration) config, query, settings, scannerFactory, startDate, endDate);
+                returnQueryData.addDelegate(results);
+                
+            }
+            
+        }
+        
+        return returnQueryData;
+    }
+    
+    private CloseableIterable<QueryData> getQueryData(ShardQueryConfiguration config, String query, Query settings, ScannerFactory scannerFactory,
+                    Date startDate, Date endDate) throws DatawaveQueryException {
+        log.debug("getQueryData in the FederatedQueryPlanner is called ");
+        CloseableIterable<QueryData> queryData;
+        ShardQueryConfiguration tempConfig = new ShardQueryConfiguration(config);
+        tempConfig.setBeginDate(startDate);
+        tempConfig.setEndDate(endDate);
+        // TODO: I think it is unnecessary to clone the DefaultQueryPlanner but I think it was requested.
+        DefaultQueryPlanner tempPlanner = new DefaultQueryPlanner(this);
+        queryData = tempPlanner.process(tempConfig, query, settings, scannerFactory);
+        return queryData;
+    }
+    
+    private TreeSet<YearMonthDay> generateStartAndEndDates(ShardQueryConfiguration configuration) {
+        
+        String startDate = DateHelper.format(configuration.getBeginDate().getTime());
+        String endDate = DateHelper.format(configuration.getEndDate().getTime());
+        
+        YearMonthDay.Bounds bounds = new YearMonthDay.Bounds(startDate, false, endDate, false);
+        
+        TreeSet<YearMonthDay> queryDates = new TreeSet<>();
+        for (ValueIndexHole valueIndexHole : configuration.getValueIndexHoles()) {
+            addDatesToSet(bounds, queryDates, valueIndexHole.getStartDate());
+            addDatesToSet(bounds, queryDates, valueIndexHole.getEndDate());
+        }
+        
+        for (FieldIndexHole fieldIndexHole : configuration.getFieldIndexHoles()) {
+            // TODO remove comparison below. calculateFieldHoles needs to be fixed.
+            if (fieldIndexHole.getStartDate().compareTo(fieldIndexHole.getEndDate()) == 0) {
+                addDatesToSet(bounds, queryDates, fieldIndexHole.getStartDate());
+                addDatesToSet(bounds, queryDates, fieldIndexHole.getEndDate());
+            }
+        }
+        
+        return queryDates;
+    }
+    
+    private void addDatesToSet(YearMonthDay.Bounds bounds, TreeSet<YearMonthDay> queryDates, String strDate) {
+        if (bounds.withinBounds(strDate))
+            queryDates.add(new YearMonthDay(strDate));
+    }
+    
+    @Override
```

Comment:
No need to override methods that simply call super

Reply:
Absolutely correct @ivakegg.  I just love it when Intellij takes away all the red stuff by autogenerating the overrides for you.  The IDE did the thinking for me and then I didn't pay attention to how ridiculous I was being.  I actually put the super.somemethod calls in the overrided functions because I was getting NPE's during testing.